internal/elasticsearch/security/system_user.go

```diff
@@ -0,0 +1,165 @@
+package security
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients"
+	"github.com/elastic/terraform-provider-elasticstack/internal/models"
+	"github.com/elastic/terraform-provider-elasticstack/internal/utils"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func ResourceSystemUser() *schema.Resource {
+	userSchema := map[string]*schema.Schema{
+		"id": {
+			Description: "Internal identifier of the resource",
+			Type:        schema.TypeString,
+			Computed:    true,
+		},
+		"username": {
+			Description: "An identifier for the system user (see https://www.elastic.co/guide/en/elasticsearch/reference/current/built-in-users.html).",
+			Type:        schema.TypeString,
+			Required:    true,
+			ForceNew:    true,
+			ValidateFunc: validation.All(
+				validation.StringLenBetween(1, 1024),
+				validation.StringMatch(regexp.MustCompile(`^[[:graph:]]+$`), "must contain alphanumeric characters (a-z, A-Z, 0-9), spaces, punctuation, and printable symbols in the Basic Latin (ASCII) block. Leading or trailing whitespace is not allowed"),
+			),
+		},
+		"password": {
+			Description:   "The user’s password. Passwords must be at least 6 characters long.",
+			Type:          schema.TypeString,
+			Optional:      true,
+			Sensitive:     true,
+			ValidateFunc:  validation.StringLenBetween(6, 128),
+			ConflictsWith: []string{"password_hash"},
+		},
+		"password_hash": {
+			Description:   "A hash of the user’s password. This must be produced using the same hashing algorithm as has been configured for password storage (see https://www.elastic.co/guide/en/elasticsearch/reference/current/security-settings.html#hashing-settings).",
+			Type:          schema.TypeString,
+			Optional:      true,
+			Sensitive:     true,
+			ValidateFunc:  validation.StringLenBetween(6, 128),
+			ConflictsWith: []string{"password"},
+		},
+		"enabled": {
+			Description: "Specifies whether the user is enabled. The default value is true.",
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     true,
+		},
+	}
+
+	utils.AddConnectionSchema(userSchema)
+
+	return &schema.Resource{
+		Description: "Updates system user's password and enablement. See, https://www.elastic.co/guide/en/elasticsearch/reference/current/built-in-users.html",
+
+		CreateContext: resourceSecuritySystemUserPut,
+		UpdateContext: resourceSecuritySystemUserPut,
+		ReadContext:   resourceSecuritySystemUserRead,
+		DeleteContext: resourceSecuritySystemUserDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
```

Comment:
I'm not sure it ever makes sense to import this resource? A user would never be able to import the password/hash and so it's pretty likely there would be changes to be applied. 

Given the 'create' doesn't actually create the user I'd be inclined to just make this not importable. WDYT? 
```suggestion
```

Reply:
Oh that's true and making the resource not importable makes sense.
Fixed it!
https://github.com/elastic/terraform-provider-elasticstack/pull/188/commits/a641bcd07ecf8a5c55f7eeef95bb52e023ce4379